### PR TITLE
feat: support external etcd

### PR DESCRIPTION
**Pull request by chunshao90**
## Rationale
Support external etcd cluster.

## Detailed Changes
* Add option to enable embed etcd cluster.
* [Breaking change] rename `default-http-port` to `http-port`.

## Test Plan
Manual test and CI.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -268,7 +330,10 @@ func (ctx *leaderWatchContext) ShouldStop() bool {
 }
 
 func (ctx *leaderWatchContext) EtcdLeaderID() uint64 {
-	return ctx.srv.etcdSrv.Server.Lead()
+	if ctx.srv.etcdSrv != nil {
+		return ctx.srv.etcdSrv.Server.Lead()
+	}
+	return 0
```
Comment (jiacai2050) [inline server/server.go:336]: what does 0 mean?

File: server/member/member.go [hunk 2]
```diff
@@ -77,7 +77,7 @@ func (m *Member) getLeader(ctx context.Context) (*getLeaderResp, error) {
 		return nil, ErrInvalidLeaderValue.WithCause(err)
 	}
 
-	return &getLeaderResp{Leader: leader, Revision: leaderKv.ModRevision, IsLocal: leader.GetId() == m.ID}, nil
+	return &getLeaderResp{Leader: leader, Revision: leaderKv.ModRevision, IsLocal: leader.GetEndpoint() == m.Endpoint}, nil
```
Comment (jiacai2050) [inline server/member/member.go:80]: - Add a method `isLocal` to encapsulate this logic.
- Why change id check to endpoint?

File: server/member/watch_leader.go [hunk 3]
```diff
@@ -26,25 +26,35 @@ type WatchContext interface {
 }
 
 type LeaderWatcher struct {
-	watchCtx    WatchContext
-	self        *Member
-	leaseTTLSec int64
+	watchCtx        WatchContext
+	self            *Member
+	leaseTTLSec     int64
+	watchEtcdLeader bool
```
Comment (jiacai2050) [inline server/member/watch_leader.go:32]: Add comments explaining when this will be false/true.

File: server/member/watch_leader.go [hunk 4]
```diff
@@ -131,3 +141,65 @@ func (l *LeaderWatcher) Watch(ctx context.Context, callbacks LeadershipEventCall
...
+		}
+
+		if wait != waitReasonNoWait {
+			logger.Warn("sleep a while during watch", zap.String("wait-reason", wait))
+			time.Sleep(watchLeaderFailInterval)
+			wait = waitReasonNoWait
+		}
+
+		// Check whether leader exists.
+		leaderResp, err := l.self.getLeader(ctx)
```
Comment (jiacai2050) [inline server/member/watch_leader.go:178]: resp

File: server/server.go [hunk 5]
```diff
@@ -89,9 +98,18 @@ func (srv *Server) Run(ctx context.Context) error {
 		return err
 	}
 
+	// If enableEmbedEtcd is false, the grpc server is started in a separate process.
+	if !srv.cfg.EnableEmbedEtcd {
```
Comment (jiacai2050) [inline server/server.go:102]: Remove this if block, add start grpc server at L90 with a else branch.

File: server/member/member.go (hunk 2 above)
Comment (chunshao90) [reply server/member/member.go:80]: The original implementation uses etcd id to determine whether it is local node. Now when using an external etcd cluster, you need to judge by the registered node address.

File: server/server.go (hunk 1 above)
Comment (chunshao90) [reply server/server.go]: return err.

**Review by ShiKaiWi — APPROVED** [review]

**Pull request merged**